core/src/main/java/io/questdb/griffin/engine/table/PageFrameRecordCursorFactory.java

```diff
@@ -106,6 +109,11 @@ public PageFrameCursor getPageFrameCursor(SqlExecutionContext executionContext,
 
     @Override
     public int getScanDirection() {
+        if (singleRowFactory) {
+            // we only return single row, sometimes we use backward scan to do that
+            // even if we do, we mark single row factory to return data in ascending timestamp order
```

Comment:
>  even if we do, we mark single row factory to return data in ascending timestamp order

why?

Reply:
there is validation in as-of and lt-join generator code, which checks that both left and right factories are in ascending order. Without this change single row symbol search will fail to participate in those joins

Reply:
Please include as code comment. The comment does not explain WHY it pretends to use forward scan for single row.

Reply:
done in https://github.com/questdb/questdb/pull/5377/commits/f7861d6fe783a696d7b819712e2f9a0ebb6faa9d